public/services/ManagedIndexService.ts

```diff
@@ -13,56 +13,73 @@ import {
 } from "../../server/models/interfaces";
 import { ServerResponse } from "../../server/models/types";
 import { NODE_API } from "../../utils/constants";
+import { MDSEnabledClientService } from "./MDSEnabledClientService";
 
-export default class ManagedIndexService {
-  httpClient: HttpSetup;
-
-  constructor(httpClient: HttpSetup) {
-    this.httpClient = httpClient;
-  }
-
-  getManagedIndex = async (managedIndexUuid: string): Promise<ServerResponse<any>> => {
-    const response = (await this.httpClient.get(`..${NODE_API.MANAGED_INDICES}/${managedIndexUuid}`)) as ServerResponse<any>;
+export default class ManagedIndexService extends MDSEnabledClientService {
+  getManagedIndex = async (managedIndexUuid: string, queryObject: HttpFetchQuery | undefined): Promise<ServerResponse<any>> => {
+    let url = `..${NODE_API.MANAGED_INDICES}/${managedIndexUuid}`;
+    const query = this.patchQueryObjectWithDataSourceId(queryObject);
+    const params = query ? { query } : {};
+    const response = (await this.httpClient.get(url, params)) as ServerResponse<any>;
```

Comment:
can we please avoid "any" type

Reply:
+1, https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#avoid-any-whenever-possible

Reply:
Done